macrosynergy/management/OAuth_connection.py

```diff
@@ -0,0 +1,98 @@
+
+"""
+DataQuery Interfaces and API wrappers.
+"""
+import requests
+import json
+from typing import Union
+# The functools module is for higher-order functions: functions that act on or return
+# other functions.
+from functools import partial
+from datetime import datetime
+
+class DataQueryOAuth(object):
+    """
+    DataQuery REST web API class
+
+    For unittesting mock up methods
+
+        1. _create_api (return: mp_api)
+        2. _renew_token (mb_api fetch_token)
+        3. _fetch (mb_api get and post)
+
+    :param <str> client_id: string with client id, username.
+    :param <str> client_secret: string with client secret, password.
+    """
+
+    def __init__(self, client_id: str, client_secret: str):
+
+        url = "https://api-developer.jpmorgan.com/research/dataquery-authe/api/v2/"
+        self.base_url = url
+
+        self.__token_url = "https://authe.jpmchase.com/as/token.oauth2"
+        self.__dq_api_resource_id = 'JPMC:URI:RS-06785-DataQueryExternalApi-PROD'
```

Comment:
Similar to above `url` for `TOKEN_URL = "https://authe.jpmchase.com/as/token.oauth2"` and `DQ_RESOURCE_ID = "JPMC:URI:RS-06785-DataQueryExternalApi-PROD"`